includes/class-monk-links.php

```diff
@@ -309,31 +319,40 @@ public function monk_add_language_home_permalink( $link, $path ) {
 	 * @return $url The changed link.
 	 */
 	public function monk_change_language_url( $url, $lang ) {
-		$monk_languages = monk_get_available_languages();
+		$monk_languages       = monk_get_available_languages();
 
-		$default_language     = get_option( 'monk_default_language', false );
+		$default_language     = $this->site_language;
 		$default_language_url = get_option( 'monk_default_language_url', false );
 		$active_languages     = $this->monk_get_active_languages();
 
 		if ( in_array( $lang, $active_languages, true ) ) {
 			$language = $lang;
-		} else {
+		} elseif ( array_key_exists( $lang, $monk_languages ) ) {
 			$language = $monk_languages[ $lang ]['slug'];
+		} else {
+			$language = $monk_languages[ $default_language ]['slug'];
 		}
 
 		if ( $this->monk_using_permalinks() ) {
-
 			if ( ! empty( $active_languages ) ) {
 				$base    = $this->site_home . '/' . $this->site_root;
-				$slug    = $default_language_url || $monk_languages[ $default_language ]['slug'] !== $language ? $language . '/': '';
+				if ( ( $default_language_url || ( ! $default_language_url && $language !== $default_language ) ) ) {
```

Comment:
Avoid simple if/else blocks, define the variable first as `$slug = null`, then cover the cases you may need.